### PR TITLE
[ac] Fix save block functionality after pipeline execution

### PR DESCRIPTION
**Pull request by anniexcheng**
# Description

There was a bug on the Pipeline Runs page where after clicking the "Execute Pipeline" or "Cancel Pipeline" button, you could no longer save a code block via the `cmd + s` shortcut ([Airtable task](https://airtable.com/applEuqJsK4zF5yJK/tbl5bIlFkk2KhCpL4/viwCmaGgcnwD9IqOM/recdr4j0pXLg8tAFc?blocks=hide)). It would instead bring up the default system file save dialog:

https://github.com/mage-ai/mage-ai/assets/8130751/5aafe36b-4825-4734-9033-094aa5a2f526

The issue was that in `SidekickContainerStyle`, we disabled keyboard shortcuts in `onFocus` but the `onBlur` event was never triggered when the sidekick focus was lost. This is a result of a `div` not being a focusable element by default. This PR fixes this issue!

# How Has This Been Tested?
- [x] Test you can use the `cmd + s` shortcut to save a code block after clicking the `Execute pipeline` button
- [x] Test you can use the `cmd + s` shortcut to save a code block after clicking the `Cancel pipeline` button
- [x] Test you can use the `cmd + s` shortcut after interacting with the dependency graph
- [x] Test you can use the `cmd + s` shortcut after clicking the other buttons/toggles in the sidekick

**Pipeline Execution**

https://github.com/mage-ai/mage-ai/assets/8130751/1a756451-0666-40d6-b374-6e491275a867

**Global Variables**

https://github.com/mage-ai/mage-ai/assets/8130751/171ae052-cc48-4357-9e12-62877dde8a8f

**Secrets**

https://github.com/mage-ai/mage-ai/assets/8130751/036aa838-a2c5-4436-9771-ae49f921134b

cc: @johnson-mage

### CODE REVIEW

File: mage_ai/frontend/components/PipelineDetail/PipelineExecution/index.tsx [hunk 1]
```diff
@@ -47,6 +47,17 @@ function PipelineExecution({
     numberOfMessages > 100 ? pipelineMessages.slice(-100) : pipelineMessages
   ), [numberOfMessages, pipelineMessages]);
 
+  // When the pipeline starts executing, the execution button gets disabled.
+  // Disabled buttons are not focusable, so manually remove the focus here.
```
Comment (anniexcheng) [inline mage_ai/frontend/components/PipelineDetail/PipelineExecution/index.tsx:51]: This was the root of the bug. We couldn't remove focus from a disabled button!

File: mage_ai/frontend/components/Sidekick/index.tsx [hunk 2]
```diff
@@ -582,6 +582,7 @@ function Sidekick({
           }
         }}
         overflowHidden={activeView === ViewKeyEnum.TREE}
+        tabIndex={0} // Make this div a focusable element
```
Comment (anniexcheng) [inline mage_ai/frontend/components/Sidekick/index.tsx:585]: Need this; otherwise, `onBlur` won't get triggered on loss of focus

**Comment by johnson-mage** [comment]
@anniexcheng This is great, thanks for solving that issue! On a related note, a similar kind of thing happens in the `Variables` and `Secrets` tabs of the Sidekick as well. Can you please fix the focus issue there as well? We can just include it in this PR as well.

Attached a gif showing the issue, but to recreate:
1. Open `Variables` or `Secrets` tab in Sidekick.
2. Create a "New" variable/secret OR edit an existing one.
3. Press "Enter" after editing the value.
4. Click in the code editor section of a block.
5. Try to save with keyboard shortcut (`cmd+S`).
6. The system save file dialogue should open instead of the pipeline saving. Note: Clicking on the Sidekick and then clicking back into the block code editor "fixes" the issue and allows you to save the pipeline with the keyboard shortcut again.

sidekick focus on save

**Comment by anniexcheng** [comment]
@johnson-mage fixed the save functionality in the Global Variables and Secrets pages too! PTAL :)

**Review by johnson-mage — APPROVED** [review]

**Pull request merged**